WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java

```diff
@@ -428,14 +428,21 @@ private void handleConnectSiteInfoForWoo(ConnectSiteInfoPayload siteInfo) {
             endProgressIfNeeded();
             // Not a WordPress site
             mLoginListener.handleSiteAddressError(siteInfo);
+        } else if (siteInfo.hasJetpack && siteInfo.isJetpackConnected && siteInfo.isJetpackActive) {
```

Comment:
The only things that I noticed are with these scenarios: 

- Jetpack is installed but the plugin is not active
- Jetpack is installed and active but not connected

In these cases, the user will receive the "The site is not a WordPress site" error.  If we only keep the `siteInfo.hasJetpack` clause, we will be able to proceed with a better error message for both scenarios:

<img src="https://user-images.githubusercontent.com/198826/135929105-00944502-afd7-4321-b6b3-4173f22fb815.png" width="300">

Do you think we should keep this as this only?

```suggestion
        } else if (siteInfo.hasJetpack) {
```

Would we be causing a bug somewhere else? 😅 I don't see any other regression when doing so but you might know more. 

I also recognize that we've probably spent a long time dealing with this and I think it's okay to stop here for now and reduce the risk. So I leave it up to you to decide. 🙂

Reply:
I'll re-test this, but if that's truly the case it's a regression introduced in this PR -> it should show "Jetpack not connected" error - more info on [the previous PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/67). Thanks for catching this, I'll look into it in the following days.

Reply:
I've retested the two mentioned scenarios and they worked as expected without any changes to the code.

- Jetpack is installed and active but not connected

| Web | App |
|-----|-----|
| <img width="2304" alt="Screenshot 2021-10-06 at 10 18 33" src="https://user-images.githubusercontent.com/2261188/136166453-a8618418-30f3-4353-bbf9-d1b879fa5cb1.png"> | <img width="516" alt="Screenshot 2021-10-06 at 10 18 43" src="https://user-images.githubusercontent.com/2261188/136166491-9e76e68b-75f7-4f97-b5dd-2ac25255f2b1.png"> |

- Jetpack is installed but the plugin is not active

| Web | App |
|-----|-----|
| <img width="2299" alt="Screenshot 2021-10-06 at 10 19 13" src="https://user-images.githubusercontent.com/2261188/136166774-06b933b6-18c8-46ab-bf7d-eb1da87ef810.png"> | <img width="519" alt="Screenshot 2021-10-06 at 10 19 18" src="https://user-images.githubusercontent.com/2261188/136166800-09737cef-124e-414f-9979-44b2c36d173e.png"> |


I also tried using the URL of your site from the screenshot `https://confidential-aardwolf.jurassic.ninja/` but it seems it already has jetpack active and connected as it redirects me to the .com login flow.

Could you please double check that you were testing the changes on the correct version of the app and share an url which is not working as expected in case you'll be able to reproduce the issue. Thank you @shiki!!!

Reply:
> In these cases, the user will receive the "The site is not a WordPress site" error. If we only keep the siteInfo.hasJetpack clause,

Also, just for future reference - IMO we can't keep just the `hasJetpack` clause as we wouldn't be able to differentiate between `Jetpack not connected` and `xmlrpc disabled` errors. When "xml-rpc" is disabled, "isJetpackConnected" returns false even Jetpack is connected => we need to run "initiateDiscovery()" in that scenario - [more info on this PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/67).

Reply:
@malinajirka It's making a bit more sense now. Thank you for those screenshots. And sorry I did not provide a correct list of steps. My bad! 

So, the difference between our testing is that I had Disable XML-RPC API plugin installed. From your screenshots, it looks like you didn't have that installed. 

<img src="https://user-images.githubusercontent.com/198826/136265333-01dbcaed-0d77-47e3-85b7-eeaf8cb4a2c4.png" width="600">


1. Install XML-RPC API plugin and activate it. 
2. Do not activate Jetpack. 
3. Follow site address flow. 
4. This is the result:
    <img width="300" alt="Pixel 3 2021-10-06 12-47-55" src="https://user-images.githubusercontent.com/198826/136265378-458fddb0-6f09-4554-b660-e220c5c5a67d.png">

If XML-RPC API plugin is not installed, then I get the same (better) error as you: 

<img width="300" alt="Pixel 3 2021-10-06 12-50-15" src="https://user-images.githubusercontent.com/198826/136265469-82b0d21d-cb6b-4a9a-bbe8-c92360b26575.png">

Reply:
> When "xml-rpc" is disabled, "isJetpackConnected" returns false and hasJetpack returns true even Jetpack is connected 

Got it. I'm seeing a different result. I tested different combinations and here is the result. See Scenario F. That's the scenario that you meant, right? 

The `isJetpackConnected` seems to be consistent and correct for me. No idea why! 😄 I tested Scenario F on both Jurassic and Pressable.

|   | Web: XML-RPC Plugin Active? | Web: Jetpack Installed? | Web: Jetpack Active? | Web: Jetpack Connected? |.isJetpackActive |.isJetpackConnected |.hasJetpack |
|---|-----------------------------|-------------------------|----------------------|-------------------------|--------------------------|-----------------------------|---------------------|
| a | -                           | Y                       | -                    | -                       | -                        | -                           | true                |
| b | -                           | Y                       | Y                    | -                       | true                     | -                           | true                |
| c | -                           | Y                       | Y                    | Y                       | true                     | true                        | true                |
| d | Y                           | Y                       | -                    | -                       | -                        | -                           | true                |
| e | Y                           | Y                       | Y                    | -                       | true                     | -                           | true                |
| f | Y                           | Y                       | Y                    | Y                       | true                     | true                        | true                |
| g | -                           | -                       | -                    | -                       | -                        | -                           | -                   |
| h | Y                           | -                       | -                    | -                       | -                        | -                           | -                   |

:shipit: But, given that you're seeing a different result, I think we can assume that it's broken somewhere and it's out of our control.

Reply:
> It's making a bit more sense now. Thank you for those screenshots. And sorry I did not provide a correct list of steps. My bad!

100% my bad, sorry. I should have tested both scenarios.

> Got it. I'm seeing a different result. I tested different combinations and here is the result. See Scenario F. That's the scenario that you meant, right?
> The isJetpackConnected seems to be consistent and correct for me. No idea why! 😄 I tested Scenario F on both Jurassic and Pressable.

Thanks so much for the table ;)! I've retested the scenarios and I can confirm the table is perfect! However, I went back to the original PR and we were testing the "disable xmlrpc" there with the `.htaaccess` file method. I've retested that scenario and 

|   | Web: xmlrpc disabled using .htaccess | Web: Jetpack Installed? | Web: Jetpack Active? | Web: Jetpack Connected? |.isJetpackActive |.isJetpackConnected |.hasJetpack |
|---|-----------------------------|-------------------------|----------------------|-------------------------|--------------------------|-----------------------------|---------------------|
| i | Y | Y | Y | Y | false | false | true | 
| i | Y | Y | Y | - | false | false | true |
| i | Y | Y | - | - | false | false | true |
| i | Y | - | - | - | false | false | false |


If we simplified the condition to `siteInfo.hasJetpack`, the check would not work for the `.htaccess` scenario. So from all the above info I think we have two options
1. Keep the current code, have good messages for .htaccess  but not so good for the disable-xml-rpc-api plugin
2. Update the code, have good messages for disable-xml-rpc-api plugin, but not so good for the .htaccess

Tbh I'm not sure which approach is more common so it's a tough decision. My reasoning is that the xml-rpc-disabled plugin is visible on the UI - on web. However, htaccess file can be created by the hosting provider and is not visible on the UI - so perhaps having a better UX for this scenario is more important, wdyt?

Either way, since this PR is approved I'll proceed with the merge. Having said, if we decide to simplify the condition, I'll be happy to create a follow-up PR ;). 

Thanks so much for the great review and all the investigation!! It's truly appreciated.

Reply:
I agree with opting for the `.htaccess` one! The plugin is manually installed so customers would be typically aware of what's wrong. 

Thank you!

Reply:
Just to close the loop - there has been a follow up discussion and we updated the tables - > https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/72#issuecomment-954657853